### ISSUE TITLE
why not just `~/.config/lazygit/config.yml`

### ISSUE THREAD

**Issue by smartding** [jesseduffield/lazygit#913]
The config file in Linux is `~/.config/jesseduffield/lazygit/config.yml`?  
Why not just `~/.config/lazygit/config.yml`?

**Comment by mjarkk** [comment]
We have discussed this in another issue but i can't remember which one it was.  
TL;DR:  
The path we now use is a mix of the xdg directory guidelines mixed with the *(old)* GoLang way of storing projects.  
I think we all agreed in that issue on the fact that following only the xdg directory guidelines would indeed be better but because we currently already use `jesseduffield/lazygit` and because the file is editable via lazygit we didn't put in the work to change change it.  

I think no one would be angry if we moved `jesseduffield/lazygit` to `lazygit` but it's rather the question who wants to make the code that makes sure we transform to this new dir and makes sure no-one has to re-write their config.  
That's probably the main reason why we still use this dir and i can't blame anyone for not implementing this yet :).

**Comment by smartding** [comment]
No one would lose their config file if you start reading config from `~/.config/jesseduffield/lazygit/config.yml`. 
No need to change the code, just state this change in your readme or changlog. It's ok if some software update breaks some stuff, as long as the fix is just moving the config somewhere else.

**Comment by mjarkk** [comment]
Hmm i think our opinions differ a bit here,  
I prefer software to not just out of nothing use another config and break the look and feel but it's not my project so @jesseduffield should decide what's the best thing to do here.

**Comment by jesseduffield** [comment]
I'm fine with changing to just having the directory as 'lazygit' but we're using a Go package at the moment called config-dir which requires specifying a vendor name and an application name. I chose 'jesseduffield' as the vendor because it matches what's on github. If somebody wants to put up a PR which stops using that package (or instead fork the package so that a vendor name need not be specified) then I'll be happy to review it, but that PR would also need to handle porting existing configs over.

**Comment by Liberatys** [comment]
Maybe I'm late to the party ^^ but I checked how the vendor is handled in the package and it seems like one could just pass an empty string as the vendor and this would then produce the expected path. This is mostly down to the package using filepath.Join to concat the given vendor and dir name. And filepath.Join ignores empty strings in concatination ^^. Tried it on the project locally and it worked ^^

**Comment by Liberatys** [comment]
And one could first check if a config.yml file can be found the previous config folder and if not check for ./config/lazygit

**Comment by smartding** [comment]
> Maybe I'm late to the party ^^ but I checked how the vendor is handled in the package and it seems like one could just pass an empty string as the vendor and this would then produce the expected path. This is mostly down to the package using filepath.Join to concat the given vendor and dir name. And filepath.Join ignores empty strings in concatination ^^. Tried it on the project locally and it worked ^^
> 
> I played a bit with it and now have implement a trivial version of it.
> 
> [Liberatys#1](https://github.com/Liberatys/lazygit/pull/1)

Why no make a PR to this repo? instead of your own fork.

**Comment by Liberatys** [comment]
Because I was unsure how jesse wanted to handle the matter and thus opted into first providing my idea before opening a pr for the issue.

But if this is an acceptable workaround for the issue I'm happy to open a pr ^^

**Comment by jesseduffield** [comment]
@Liberatys I've taken a look at your PR and I've left a comment :) Feel free to raise it as a PR here as well

**Comment by Liberatys** [comment]
@jesseduffield Thank you for taking the time :D I opened the pr and made a change to address your comment.

**Comment by jesseduffield** [comment]
This is now resolved thanks to @Liberatys ! If you move your config folder up on directory it'll all work fine (we're maintaining backwards compatibility for people with the old config location, but all new lazygit uses will have the config directory created without the jesseduffield/ parent directory. Let me know if you have any issues :)
